application/clicommands/ProcessCommand.php

```diff
@@ -150,15 +160,34 @@ protected function renderProblemTree($tree, $useColors = false, $depth = 0)
             if ($useColors) {
                 $state = $this->screen->colorize($state, $colors[0], $colors[1]);
             }
-
-            $output .= sprintf(
-                "%s%s %s %s\n",
-                str_repeat('  ', $depth),
-                $node instanceof BpNode ? $node->getOperator() : '-',
-                $state,
-                $node->getAlias()
-            );
-            $output .= $this->renderProblemTree($subtree['children'], $useColors, $depth + 1);
+            if ($htmlcolors) {
+                $state = substr($state, 1, strlen($state)-2);
+                $lstate = strtolower($node->getStateName());
+                $config = $this->Config();
+                FakeRequest::setConfiguredBaseUrl($config->get("Main", "baseurl"));
+                $output.="<table class=''><tbody><tr>"
+                .str_repeat('<td></td><td></td><td></td>', $depth)
+                ."<td>";
+                if ($node instanceof BpNode) {
+                    $output.=$node->getOperator();
+                } else {
+                    $output.="-";
+                }
+                    $output.="</td><td><div class='badge state-$lstate'>$state</div> ";
+                    $node instanceof BpNode ?
+                    $output.=$node->getAlias() :
```

Comment:
Please use `$output .= $x ? $a : $b` instead of `$x ? $output .= $a : $output .= $b`.